webapp/packages/game/src/index.ts

```diff
@@ -82,7 +83,24 @@ export const OpenStarTerVillage: Game<type.State.Root> = {
         moves: {
           createProject: () => { },
           recruit: () => { },
-          contribute: () => { },
+          contribute: (G, ctx, contributions: { id: number; slotId: number; value: number; }[]) => {
+            const activeProjects = G.table.activeProjects
+            const isInvalid = contributions.map(({ id, slotId }) => {
+              if (!(0 <= id && id < activeProjects.length)) {
+                return true;
+              }
+              if (activeProjects[id].slots[slotId] === 0) {
+                return true;
+              }
+            }).some(x => x);
+            if (isInvalid) {
+              return INVALID_MOVE;
+            }
+
+            contributions.forEach(({ id, slotId, value }) => {
+              activeProjects[id].slots[slotId] = Math.max(6, activeProjects[id].slots[slotId] + value);
```

Comment:
I guess you would like to use Math.min to make sure the maximum contribution is 6 here?

Reply:
you are correct. Thanks for finding out!